Resources/application.py

```diff
@@ -832,6 +854,27 @@ def addCustomWebService(self, stationDict):
 
         return
 
+    def appendDatasetDictionaryItem(self, dataID, stationType, stationNumber, stationName, stationParam, stationUnits, resamplingMethod, decodeOptions):
+        """
+        This function adds an entry to the datasetDictionary and adds it to the Station Tab table
+        """
+        duplicateDataset = False
+        #Add validation code to see if dataset already exists in datasetDirectory
+
+        for dCounter in range(0, len(self.datasetDirectory['datasets'])):
+            if self.datasetDirectory['datasets'][dCounter]["TYPE"] == stationType and self.datasetDirectory['datasets'][dCounter]["ID"] == stationNumber and self.datasetDirectory['datasets'][dCounter]["Name"] == stationName and self.datasetDirectory['datasets'][dCounter]["Parameter"] == stationParam and self.datasetDirectory['datasets'][dCounter]["Units"] == stationUnits:
+                duplicateDataset = True
+                break
+           
+        if not duplicateDataset:
+            self.datasetDirectory['datasets'].append({"PYID":dataID,"TYPE":stationType,"ID":stationNumber,"Name":stationName,"Parameter":stationParam,"Units":stationUnits,"Resampling":resamplingMethod,"Decoding":decodeOptions, "Data":{}, "lastDateTime":None})
+            self.stationsTab.stationInfoPane.stationTable.addRow([dataID, stationType, stationNumber, stationName, stationParam])
+        else:
+            button = QtWidgets.QMessageBox.question(self, 'Error','Dataset has already been selected...'.format(traceback.format_exc()), QtWidgets.QMessageBox.Ok)
```

Comment:
Do we even need to display an error message here. It should be very obvious that the user tried to add a duplicate station.

Reply:
would be nice to have some kind of feedback as to why the GUI isn't doing anything when a user is attempting to add a duplicate. with that said, i don't care either way whether we have a pop-up here or not.

Reply:
Let's leave this the way it is for now and I'll merge the branches. Tomorrow I plan to add in my changes to the station tab which already include a similar functionality using the QAbstractAnimation library.

Reply:
awesome sauce. i'll commit the additional T1 validation and leave this as is then. thanks!